Client/Frontend/Browser/PasswordHelper.swift

```diff
@@ -171,4 +172,95 @@ class PasswordHelper: BrowserHelper {
             })
         })
     }
+
+    func handleAuthRequest(viewController: UIViewController, challenge: NSURLAuthenticationChallenge, completion: (password: Password?) -> Void) {
+        // If there have already been too many login attempts, we'll just fail.
+        if challenge.previousFailureCount > PasswordHelper.MaxAuthenticationAttempts {
+            completion(password: nil)
+            return
+        }
+
+        var credential = challenge.proposedCredential
+
+        // If we were passed an initial set of credentials from iOS, try and use them.
+        if let proposed = credential {
+            if let user = proposed.user {
+                if !user.isEmpty {
```

Comment:
You can simplify these two statements into one:

```
if !(proposed.user?.isEmpty ?? true) {
```

Personally, I'd also move this statement outside of the `if let proposed = credential` block to remove a level of nesting, and then we'd just skip over the `if let proposed = credential` block since `credential` would be `nil`...but your call.